src/pydvl/influence/frameworks/torch_differentiable.py

```diff
@@ -47,6 +203,11 @@ def stack(a: Sequence[torch.Tensor], **kwargs):
     return torch.stack(a, **kwargs)
 
 
+def cat(a: Sequence[torch.Tensor], **kwargs):
```

Comment:
Is this really needed?

Reply:
it is created (like many of the methods here e.g. stack) to keep general.py framework independent...

Reply:
I see.

How about using [singledispatch](https://docs.python.org/3/library/functools.html#functools.singledispatch) from functools instead?

You could have a generic implementation that does nothing and then specialized implementations for each framework.

For example, in separate module (maybe named `common`) inside the frameworks package we could define:

```python
from functools import singledispatch
from typing import Any

@singledispatch
def cat(a: Any) -> Any:
    """Concatenates a sequence of arrays into array"""
    raise RunetimeError(f"Array of type {type(a)} is not supported")
```

And then inside the torch_differentiable module we would do:

```python
import torch

from .common import cat

@cat.register
def cat(a: torch.Tensor) -> torch.Tensor:
    """Concatenates a sequence of tensors into a single torch tensor"""
    return torch.cat(a, **kwargs)
```

This would simplify a bit the `__init__.py` module of the frameworks package.
You would only need to expose the functions from the common module.

You would still need the import guard from the different framework modules though. So it's not a perfect solution, just a small improvement.

P.S. The code is untested (it should work tough) and is just a suggestion.

Reply:
Yeah the __init__ is a mess at the moment, but I am leaving a better handling of it for another time, or I will go crazy. Will open an issue with this suggestion =)